main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DefCompleter.scala

```diff
@@ -53,4 +58,18 @@ object DefCompleter {
     val isRoot = decl.sym.namespace.isEmpty
     isRoot || isResolved
   }
+
+  /**
+    * Returns the expression context at the given `uri` and position `pos`.
+    */
+  private def getExprContext(uri: String, pos: Position)(implicit root: Root, flix: Flix): ExprContext = {
+    val stack = LspUtil.getStack(uri, pos)
+    // The stack contains the path of expressions from the leaf to the root.
+    stack match {
+      case Expr.Error(_, _, _) :: Expr.ApplyClo(_, _, _, _, _) :: _ =>
```

Comment:
Should there be ApplyDef and ApplySig here too?

Reply:
A bit more documentation of what's going on (an example) would be nice.

Reply:
I am leaving all of this for @chanattan